data-pipeline/src/trace_exporter.rs

```diff
@@ -1,19 +1,124 @@
-use crate::agent_client::AgentClient;
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
 
-pub struct TraceExporter {
-    host: String,
-    port: u16,
-    client: AgentClient
+use bytes::Bytes;
+use datadog_trace_protobuf::pb;
+use datadog_trace_utils::trace_utils::{self, SendData, TracerHeaderTags};
+use ddcommon::{connector, Endpoint};
+use futures::executor::block_on;
+use hyper::{Body, Client, Method};
+use log::error;
+use std::{collections::HashMap, ops::Deref, str::FromStr};
+
+pub struct TraceExporter<'a> {
+    endpoint: Endpoint,
+    tracer_header_tags: TracerHeaderTags<'a>,
+    no_proxy: bool,
 }
 
-impl TraceExporter {
+impl TraceExporter<'_> {
     pub fn builder() -> TraceExporterBuilder {
         TraceExporterBuilder::default()
     }
 
-    pub fn send(&mut self, data: Vec<u8>, trace_count: usize) -> Result<String, String> {
-        let url = format!("{}:{}{}", self.host, self.port, "/v0.4/traces");
-        self.client.blocking_send_trace(&url, data, trace_count)
+    pub fn send(self, data: Bytes, trace_count: usize) -> Result<String, String> {
+        if self.no_proxy {
+            self.send_deser_ser(data)
+        } else {
+            self.send_proxy(data, trace_count)
+        }
+    }
+
+    fn send_proxy(self, data: Bytes, trace_count: usize) -> Result<String, String> {
+        let uri = self.endpoint.url.clone();
+        block_on(async {
```

Comment:
I'm not sure that guaranteed works? `hyper` futures need to be polled on a thread that has a tokio runtime activated, but if it's a same_thread runtime, that will deadlock, because you're blocking the runtime from advancing.
If you look at what the profiler does, they hold a `tokio::Runtime` in the profiling::Exporter struct and use with spawn_blocking to run async functions

Reply:
Thanks. I'll fix that.